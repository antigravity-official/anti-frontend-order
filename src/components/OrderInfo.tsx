import type { FC } from "react";
import { Order } from "../Models";
import OrderProductLists from "./OrderProductLists";
import ShippingInfo from "./ShippingInfo";
import OrderPaymentInfo from "./OrderPaymentInfo";

interface Props {
  order: Order;
}

const OrderInfo: FC<Props> = ({ order }) => {
  return (
    <div>
      <OrderPaymentInfo paymentInfo={order.paymentInfo} />
      <p>----------------------------</p>
      <OrderProductLists productLists={order.products} />
      <p>----------------------------</p>
      <ShippingInfo shippingInfo={order.shipping} />
    </div>
  );
};

export default OrderInfo;
