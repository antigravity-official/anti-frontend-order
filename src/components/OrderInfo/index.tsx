import { OrderInfo as OrderInfoTypes } from '../../types/ordersTypes'

interface OrderInfoProps {
  orderInfoData: OrderInfoTypes
}

const OrderInfo = ({ orderInfoData }: OrderInfoProps) => {
  return (
    <>
      <p>주문번: {orderInfoData.id}</p>
      {/* <p>주문일: {orderInfo.orderAt}</p> */}
      <p>총 결제금액: ${orderInfoData.amount}원</p>
    </>
  )
}

export default OrderInfo
