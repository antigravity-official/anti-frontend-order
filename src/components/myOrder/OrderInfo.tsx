import React from 'react';
import { Order } from '../../models/Order';

const OrderInfo: React.FC<{ order: Order }> = ({ order }) => (
  <div>
    <div>주문번호: {order.id}</div>
    <div>주문일: {order.orderAt.toLocaleString()}</div>
    <div>총 결제금액: {order.amount}원</div>
  </div>
);

export default OrderInfo;
