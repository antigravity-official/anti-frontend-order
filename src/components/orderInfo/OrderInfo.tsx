import * as T from "../../type/Models";
import ProductInfo from "./ProductInfo";
import ShippingInfo from "./ShippingInfo";

interface OrderInfoProps {
  data: T.Order | null;
}

export default function OrderInfo({ data }: OrderInfoProps) {
  let output: string[] = [];
  const println = (text: string) => output.push(text);

  println(`주문번호: ${data?.id}`);
  println(`주문일: ${data?.orderAt}`);
  println(`총 결제금액: ${data?.amount}원`);
  println(``);
  println(`----------------------------`);

  return (
    <div>
      {output.map((line) => (
        <div>{line}</div>
      ))}
      {data?.products && <ProductInfo data={data.products} />}
      {data?.shipping && <ShippingInfo data={data.shipping} />}
    </div>
  );
}
