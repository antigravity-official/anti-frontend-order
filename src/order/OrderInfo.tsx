import React, {
  Fragment
} from 'react';
import OrderInfoLine from './row/OrderInfoLine'

type OrderInfoProps = {
  orderInfo:string[]
}

function OrderInfo({ orderInfo }: OrderInfoProps) {
  const nodes = orderInfo.map((
    line:string, index:number
  ) => <OrderInfoLine key={`line-${index}`} line={line} />);
  
  return (
    <Fragment>
      {nodes}
    </Fragment>
  )
}

export default OrderInfo;
