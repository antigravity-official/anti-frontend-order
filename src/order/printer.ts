import {
  Order, OrderProduct, Shipping
} from "../Models";
import {
  moneyFormatter,
  datetimeFormatter
} from '../common/util/formatter';

export function printOrder(
  order: Order
): string[] {

  let output: string[] = [];
  const println = (text: string) => output.push(text);

  println(`주문번: ${order.id}`);
  println(`주문일: ${datetimeFormatter(order.orderAt)}`);
  println(`총 결제금액: ${moneyFormatter(order.amount)}원`);
  println('----------------------------');

  printShippings(order.shippings, println);

  return output;
}

export function printShippings(
  shippings: Shipping[],
  println: (text: string) => void
) {

  return shippings.forEach((shipping: Shipping, index: number) => {
    println(`[배송정보${index + 1}]`);
    printShipping(shipping, println);
    println('----------------------------');
  });
}

export function printShipping(
  shipping: Shipping,
  println: (text: string) => void
) {
  println(`송장번호: ${shipping.trackingNumber}`);
  println(`배송료: ${moneyFormatter(shipping.shippingFee)}원`);
  println(`주소: [${shipping.post}] ${shipping.address}`);
  println(`메시지: ${shipping.message}`);
  println('');
  printProducts(shipping.products, println);
};

export function printProducts(
  products: OrderProduct[],
  println: (text: string) => void
) {
  println('[상품목록]');

  products.forEach((product: OrderProduct, index: number) => {
    printProduct(product, println);
    if (index + 1 < products.length) {
      println('');
    }
  });
};

export function printProduct(
  product: OrderProduct,
  println: (text: string) => void
) {
  println(`상품명: ${product.name}`);
  println(`가격: ${moneyFormatter(product.price)}원`);
  println(`주문정보: ${product.stock.color}/${product.stock.band}/${product.stock.cup} ${product.stock.quantity}개`);
}