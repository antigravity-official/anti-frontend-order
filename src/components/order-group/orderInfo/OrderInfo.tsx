import React from "react";
import "./OrderInfo.css";

const OrderInfo = () => {
  return <div>OrderInfo</div>;
};

export default OrderInfo;
