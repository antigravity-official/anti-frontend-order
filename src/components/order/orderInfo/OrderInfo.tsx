import React from "react";
import "./OrderInfo.scss";

const OrderInfo = () => {
  return <div>OrderInfo</div>;
};

export default OrderInfo;
