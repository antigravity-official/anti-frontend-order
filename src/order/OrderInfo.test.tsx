import React from 'react';
import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import '@testing-library/jest-dom'
import OrderInfo from './OrderInfo';
import { printOrder } from './printer';
import { Order } from '../Models';
import assetOrder from '../assets/order.json';


describe('<OrderInfo />', () => {
  
  it('스냅샷과 일치하는지 확인한다.', () => {
    const json = assetOrder
    const order: Order = {
      id: json.id,
      orderAt: new Date(json.orderAt),
      amount: json.amount,
      shippings: json.shippings
    }
    const orderInfo = printOrder(order)
    const utils = render(<OrderInfo orderInfo={orderInfo} />);
    expect(utils.container).toMatchSnapshot();
  })
})