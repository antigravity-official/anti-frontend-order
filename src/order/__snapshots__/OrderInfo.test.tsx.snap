// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderInfo /> 스냅샷과 일치하는지 확인한다. 1`] = `
<div>
  <div>
    주문번: 100
  </div>
  <div>
    주문일: 2023/01/01 일요일 12:13:14
  </div>
  <div>
    총 결제금액: 77,700원
  </div>
  <div>
    ----------------------------
  </div>
  <div>
    [배송정보1]
  </div>
  <div>
    송장번호: T08912739410
  </div>
  <div>
    배송료: 3,000원
  </div>
  <div>
    주소: [06106] 서울시 강남구 논현로 650-1
  </div>
  <div>
    메시지: 안전한 배송 부탁드립니다.
  </div>
  <br />
  <div>
    [상품목록]
  </div>
  <div>
    상품명: 더핏브라
  </div>
  <div>
    가격: 42,800원
  </div>
  <div>
    주문정보: BLACK/M(75)/A 1개
  </div>
  <br />
  <div>
    상품명: 소울 라이트 브라
  </div>
  <div>
    가격: 34,900원
  </div>
  <div>
    주문정보: SKIN/S(65~70)/B 1개
  </div>
  <div>
    ----------------------------
  </div>
  <div>
    [배송정보2]
  </div>
  <div>
    송장번호: T08912739411
  </div>
  <div>
    배송료: 3,000원
  </div>
  <div>
    주소: [06106] 서울시 강남구 논현로 650-2
  </div>
  <div>
    메시지: 부재시 경비실에 맡겨주세요.
  </div>
  <br />
  <div>
    [상품목록]
  </div>
  <div>
    상품명: 더핏브라
  </div>
  <div>
    가격: 42,800원
  </div>
  <div>
    주문정보: BLACK/M(75)/A 1개
  </div>
  <br />
  <div>
    상품명: 소울 라이트 브라
  </div>
  <div>
    가격: 34,900원
  </div>
  <div>
    주문정보: SKIN/S(65~70)/B 1개
  </div>
  <div>
    ----------------------------
  </div>
</div>
`;
